### PR TITLE
chore: Miscellaneous documentation fixes

### PR DESCRIPTION
**Pull request by twpayne**
cc @KapJI

### CODE REVIEW

File: internal/cmd/cmd.go [hunk 1]
```diff
@@ -159,18 +159,18 @@ func extractHelp(command string, data []byte, longHelpTermRenderer, exampleTermR
...
+	shortFlags := chezmoiset.New[string]()
 
 	stateChange := func(line string, state *stateType) bool {
 		switch {
-		case strings.HasPrefix(line, "## Flags") || strings.HasPrefix(line, "## Common flags"):
+		case line == "## Flags" || line == "## Common flags":
 			*state = stateInOptions
 			return true
-		case strings.HasPrefix(line, "## Example"): // accept singular and plural
+		case line == "## Example":
```
Comment (KapJI) [inline internal/cmd/cmd.go:170]: My idea was to support `## Example` or `## Note` section if someone makes a typo or intentionally uses singular form.
```suggestion
		case line == "## Examples":
```

File: assets/chezmoi.io/docs/reference/commands/add.md [hunk 2]
```diff
@@ -102,6 +102,6 @@ $ chezmoi add ~/.oh-my-zsh --exact --recursive
 
     `chezmoi add` will fail if the entry being added is in a directory
     implicitly created by an
-    [external](../special-files-and-directories/chezmoiexternal-format.md).
+    [external](../special-files/chezmoiexternal-format.md).
```
Comment (KapJI) [inline assets/chezmoi.io/docs/reference/commands/add.md:105]: I wonder if there is a way to verify that all links are valid automatically.

File: assets/chezmoi.io/mkdocs.yml [hunk 3]
```diff
@@ -350,11 +350,6 @@ markdown_extensions:
 - admonition
 - meta
 - pymdownx.details
-- pymdownx.snippets:
-    base_path:
-      - assets/chezmoi.io/snippets
-      - snippets
-    check_paths: true
```
Comment (KapJI) [inline assets/chezmoi.io/mkdocs.yml:357]: You didn't like the idea of using snippets or just wanted to write different descriptions there? I wanted to use snippets to deduplicate flags descriptions for commands as well.

File: assets/chezmoi.io/docs/reference/commands/add.md (hunk 2 above)
Comment (KapJI) [reply assets/chezmoi.io/docs/reference/commands/add.md:105]: I think I didn't pay enough attention to the `serve-docs` console. Probably worth enabling [`strict`](https://www.mkdocs.org/user-guide/configuration/#strict) in mkdocs.

File: internal/cmd/cmd.go (hunk 1 above)
Comment (twpayne) [reply internal/cmd/cmd.go]: Good catch, thanks!

File: assets/chezmoi.io/docs/reference/commands/add.md (hunk 2 above)
Comment (twpayne) [reply assets/chezmoi.io/docs/reference/commands/add.md:105]: Great spot. Added in 69ced411df2f173a6cbf4815273e10b3268d442c.

File: assets/chezmoi.io/mkdocs.yml (hunk 3 above)
Comment (twpayne) [reply assets/chezmoi.io/mkdocs.yml:357]: I wanted different descriptions there, and also want to use snippets to de-duplicate flag descriptions for commands.

The tricky might be getting the embedded Markdown parser to understand snippets.

**Review by halostatue — COMMENTED** [review]
LGTM

**Review by KapJI — APPROVED** [review]

File: assets/chezmoi.io/mkdocs.yml [hunk 4]
```diff
@@ -6,6 +6,7 @@ copyright: Copyright © Tom Payne 2018-2024
 repo_name: twpayne/chezmoi
 repo_url: https://github.com/twpayne/chezmoi
 edit_uri: edit/master/assets/chezmoi.io/docs/
+strict: true
```
Comment (KapJI) [inline assets/chezmoi.io/mkdocs.yml:9]: Let's add more strictness and also enable validation for links with anchors and link in invalid format. See recommended settings in https://www.mkdocs.org/user-guide/configuration/#validation
```suggestion
strict: true

validation:
  omitted_files: warn
  absolute_links: relative_to_docs
  unrecognized_links: warn
  anchors: warn
```

**Pull request merged**